python_ta/reporters/plain_reporter.py

```diff
@@ -112,10 +117,63 @@ def sort_messages(self):
         for msg in self._style_messages:
             self._sorted_style_messages[msg.msg_id].append(msg)
 
-    def show_file_linted(self, filename):
-        print('*'*15, 'File:', filename)
+    def set_output_stream(self, output_filepath=None):
+        """Determine where to output pyta messages.
+        • Reset a file for outputting messages into, and store its file object.
+        • Default stream to std out.
+        • Note: leave the file open during the execution of pyta program 
+        because many writes may happen, and the fd should close automatically 
+        by the system when the program ends. 
+        • Raises IOError.
+        """
+        if output_filepath is None:
+            # Stream to std out. Use method instead of setting self.out directly
+            self.set_output(sys.stdout)
+            return
+        
+        # Paths may contain system-specific or relative syntax, e.g. `~`, `../`
+        correct_path = os.path.expanduser(output_filepath)
+        if not os.path.exists(os.path.dirname(correct_path)):
+            raise IOError('path {} does not exist.'.format(output_filepath))
+        if os.path.isdir(correct_path):
+            correct_path = os.path.join(correct_path, self._output_file_name)
+        with open(correct_path, 'w') as _:  # erase file, and close it.
+            pass
+        # Use this file object to append messages.
+        self.set_output(open(correct_path, 'a'))
```

Comment:
This is better, but you should only store the filename, and open the file when absolutely necessary (at the start of `print_messages`). In general, don't store file handles as attributes.